org.eclipse.scanning.malcolm.core/src/org/eclipse/scanning/malcolm/core/AbstractMalcolmDevice.java

```diff
@@ -114,17 +115,17 @@ protected void setTemplateBean(MalcolmEventBean bean) {
 	public void start(final IPosition pos) throws ScanningException, InterruptedException {
 
 		final List<Throwable> exceptions = new ArrayList<>(1);
-		final Thread thread = new Thread(new Runnable() {
-			@Override
-			public void run() {
-				try {
-					AbstractMalcolmDevice.this.run(pos);
-				} catch (Exception e) {
-					e.printStackTrace();
-					exceptions.add(e);
-				}
+
+		final Runnable runPosition = () -> {
+			try {
+				AbstractMalcolmDevice.this.run(pos);
+			} catch (Exception e) {
+				e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)